daemon/logger/fluentd/fluentd.go

```diff
@@ -84,6 +88,15 @@ func New(info logger.Info) (logger.Logger, error) {
 		return nil, err
 	}
 
+	writeTimeout := defaultWriteTimeout
+	if info.Config[writeTimeoutKey] != "" {
+		wtd, err := time.ParseDuration(info.Config[writeTimeoutKey])
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Should we check for negative durations and produce an error?

Reply:
Also, we need to check what status-code is returned by the API in case of an invalid value (haven't checked yet) (make sure it doesn't return a 5xx error)

Reply:
We should also check for extremely long timeout values to prevent daemon waiting on the write.

Reply:
So @anusha-ragunathan how long do you thing it should wait. I was thinking of something in minutes.

Reply:
@thaJeztah Actually I am completely new to these things so I am kind of learning how can I handle server error(5xx). It will take time but I will push the changes by next week.  Actually I am bit busy with my college exams :see_no_evil:

Reply:
@dungeonmaster18 I think `logger` has `ring buffer` for this case. @cpuguy83

Reply:
As far I know ring buffer is nothing but the circular buffer with fixed size.  Correct me if I am wrong @ripcurld0

Reply:
@dungeonmaster18 : regarding setting an upper limit on the timeout, lets skip that. Users should be aware of what they are setting for the timeout value. It's hard to predict a good max value. Also, I checked other timeouts. Other than disallowing negative values, there's no other check.

Reply:
For the status-code; The API uses this code https://github.com/moby/moby/blob/master/api/server/httputils/errors.go#L33-L73 (and code from the errdefs package: https://github.com/moby/moby/tree/master/api/errdefs) to convert errors from the daemon into the correct HTTP status codes. You can try creating an API request with an incorrect value for the timeout, and check what status code is returned.